src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/SerializationRecordId.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Formats.Nrbf.Utils;
+using System.IO;
+using System.IO.Hashing;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Formats.Nrbf;
+
+/// <summary>
+/// The ID of <see cref="SerializationRecord" />.
+/// </summary>
+public readonly struct SerializationRecordId : IEquatable<SerializationRecordId>
+{
+#pragma warning disable CS0649 // the default value is used on purpose
+    internal static readonly SerializationRecordId NoId;
+#pragma warning restore CS0649
+
+    internal readonly int _id;
+
+    private SerializationRecordId(int id) => _id = id;
+
+    internal static SerializationRecordId Decode(BinaryReader reader)
+    {
+        int id = reader.ReadInt32();
+
+        if (id == 0)
```

Comment:
`0` is the only illegal value. It allowed me to avoid the need of introducing `Null` property, as all `IsNull` checks are now `Equals(default)`

I know it's less pretty, but this is the API that got approved and TBH 99.99% of users should not use it at all.